Gems/EMotionFX/Code/EMotionFX/Pipeline/RCExt/Motion/MotionDataBuilder.cpp

```diff
@@ -258,6 +265,16 @@ namespace EMotionFX
                     continue;
                 }
 
+                const SceneContainers::SceneGraph::NodeIndex boneNodeIndex = graph.Find(it->first.GetPath());
+                const char* nodeName = it->first.GetName();
+                const char* nodePath = it->first.GetPath();
+
+                // Add root joint to motion data when root motion extraction rule exsits.
+                if (rootMotionExtractionRule && boneNodeIndex == rootBoneNodeIndex)
+                {
+                    rootJointDataIndex = motionData->AddJoint(it->first.GetName(), Transform::CreateIdentity(), Transform::CreateIdentity());
```

Comment:
you declared a variable for this`it->first.GetName()` could you replace it with nodeName?

Reply:
sure good catch